### PR TITLE
Implemented Validator

### LINKED ISSUE #39: Add Validator in Login Form
The Authentication form right now does not provide any validation for the form.
Add Validation in `ui/Auth/LoginPage/Auth.dart` (there is a TODO task) and validators are present in `constants/validators.dart`.  Use the right regex for the email.
If you wanna take up this issue let me know.

### PR DESCRIPTION
**Pull request by VanshKela**
Fixed Issue https://github.com/therealansh/iiit_sonepat_app/issues/39

### CODE REVIEW

File: lib/constants/validators.dart [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
     return 'Password must be longer than 8 characters';
   } else {
     return null;
   }
-}
+};
 
-String emailValidator(String value) {
+Function(String) emailValidator = (String value) {
   Pattern pattern =
```
Comment (therealansh) [inline lib/constants/validators.dart:10]: this regex needs to be changed according to the college email id to check wether the mail id is a valid college id or not.

File: lib/ui/Auth/Login_Page/Auth.dart [hunk 2]
```diff
@@ -219,13 +222,19 @@ class _AuthState extends State<Auth> {
                                       new BorderRadius.circular(30.0),
                                 ),
                                 onPressed: () {
+                                  if (!formAuthKey.currentState.validate()) {
+                                    ScaffoldMessenger.of(context)
+                                        .showSnackBar(SnackBar(content: Text('Processing Data')));
+                                  }
                                   //[TODO] ADD VALIDATOR HERE
```
Comment (therealansh) [inline lib/ui/Auth/Login_Page/Auth.dart]: remove this comment if the task is done.

**Comment by VanshKela** [comment]
Should i close this pull request and make another one?

**Comment by VanshKela** [comment]
Changed the regex to "^[A-Za-z0-9._%+-]+@iiitsonepat.ac.in$"

**Comment by therealansh** [comment]
@VanshKela try using this regex ```/^[A-Za-z]+\.[0-9]+(@iiitsonepat)\.(ac)\.(in)$/g``` try out this site https://regex101.com/ for creating regex.

File: lib/constants/validators.dart [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
-      r'^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$';
+      r'^[A-Za-z0-9._%+-]+@iiitsonepat.ac.in$';
```
Comment (therealansh) [inline lib/constants/validators.dart]: Regex takes in any alphanumeric followed by iiitsonepat.ac.in so its a wrong regex

**Review by therealansh — CHANGES_REQUESTED** [review]

**Comment by VanshKela** [comment]
I have set autoValdiateMode to always for both email and password and changed regex again.

File: lib/ui/Auth/Login_Page/Auth.dart [hunk 4]
```diff
@@ -219,13 +224,18 @@ class _AuthState extends State<Auth> {
                                       new BorderRadius.circular(30.0),
                                 ),
                                 onPressed: () {
-                                  //[TODO] ADD VALIDATOR HERE
+                                  if (!formAuthKey.currentState.validate()) {
+                                    ScaffoldMessenger.of(context)
```
Comment (therealansh) [inline lib/ui/Auth/Login_Page/Auth.dart]: Use Fluttertoast (already in the pubsec.yaml), it looks aesthetic as well as goes with the UI

**Review by therealansh — CHANGES_REQUESTED** [review]

**Review by therealansh — APPROVED** [review]

**Comment by therealansh** [comment]
LGTM, Thanks @VanshKela for the PR.

**Comment by VanshKela** [comment]
Finally done with this.

**Pull request merged**